codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/CborSerializerGenerator.kt

```diff
@@ -402,7 +402,7 @@ class CborSerializerGenerator(
         rust(
             """
             encoder.array(
-                (${context.valueExpression.asRef()}).len().try_into().expect("`usize` to `u64` conversion failed")
+                (${context.valueExpression.asValue()}).len().try_into().expect("`usize` to `u64` conversion failed")
```

Comment:
I don't think this is right. `context.valueExpression` should be a reference here (e.g. a `&Vec<T>`), there's no need to dereference it (with `*`) to call `len()` on it.